src/utils/orgUtils.ts

```diff
@@ -65,6 +72,98 @@ export class OrgUtils {
         }
     }
 
+    public static async getCompactLayoutsForSObject(
+        sObjectName: string
+    ): Promise<Object> {
+        const org = await Org.create();
+        const conn = org.getConnection();
+
+        const result = await conn.request<Object>(
+            `/services/data/v59.0/sobjects/${sObjectName}/describe/compactLayouts`
+        );
+
+        return Promise.resolve(result);
+    }
+
+    public static async getCompactLayoutForSObject(
+        sObjectName: string,
+        recordTypeId: string
+    ): Promise<Object> {
+        const org = await Org.create();
+        const conn = org.getConnection();
+
+        const result = await conn.request<Object>(
+            `/services/data/v59.0/sobjects/${sObjectName}/describe/compactLayouts/${recordTypeId}`
+        );
+
+        return Promise.resolve(result);
+    }
+
+    public static async getCompactLayoutFieldsForSObject(
+        sObjectName: string
+    ): Promise<CompactLayoutField[]> {
+        try {
+            const fields: CompactLayoutField[] = [];
+
+            // Get all the compact layouts associated to this sObject first
+            let result = await this.getCompactLayoutsForSObject(sObjectName);
+
+            if (result) {
+                // sObject can have multiple compact layouts associated with it. Get the default.
+                const defaultCompactLayoutId =
+                    result['defaultCompactLayoutId' as keyof Object];
+
+                // Mapping tab
+                const recordTypeCompactLayoutMappings =
+                    result['recordTypeCompactLayoutMappings' as keyof Object];
```

Comment:
Nit: `as keyof Object` here doesn't really offer anything, and is actually somewhat misleading. `as keyof <SomeType>` is meant to offer a type assertion that the property name falls under the explicit list of properties defined by `SomeType`. It's basically another form of a type check on the named property you're giving in the square bracket notation.

In this case, `keyof Object` doesn't really make sense, because `Object` doesn't specify any of these property names in its definition—they're ours. It seems to be happily working anyway, but it's misleading nomenclature in this case. See the [`keyof` docs](https://www.typescriptlang.org/docs/handbook/2/keyof-types.html) for more information.

Reply:
Without `as keyof Object` this is what I get in the editor.
<img width="888" alt="image" src="https://github.com/salesforce/salesforcedx-vscode-mobile/assets/42476999/fa4b4987-2a9a-4422-a400-ec94ee1e5deb">